### PR TITLE
[SP-144] Added database migrator

### PR DESCRIPTION
**Pull request by klebedev**

### CODE REVIEW

File: scripts/dev/add_migration.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (Ne4istb) [inline scripts/dev/add_migration.sh]: add_migration.bat is needed as most of the developers are working on windows :(

File: server/lib/src/database/database_migrator.dart [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  Future<Connection> connect(){
+    print("Establishing DB connection...");
+    return new DatabaseConnection().connect();
+  }
+
+  Future<Connection> ensureAppliedScriptsExists(Connection conn) async{
+    var query = "SELECT 1 FROM information_schema.tables WHERE table_schema = 'public' AND table_name = 'appliedscripts'";
+
+    await conn.query(query).isEmpty.then((isEmpty) async {
```
Comment (Ne4istb) [inline server/lib/src/database/database_migrator.dart]: This part could be simplified like
```dart  
var isEmpty = await conn.query(query).isEmpty;

if (isEmpty) 
  await CreateAppliedScriptsTable(conn);
else
  print("AppliedScripts table found!");
   
return conn;
```

The same in getSingleMigrationPromise

File: scripts/dev/add_migration.sh (hunk 1 above)
Comment (klebedev) [inline scripts/dev/add_migration.sh]: I am в курсе of this. 
Will add .bat files in the upcoming days. Have a bit of a problem, cause of not having machine on Windows.

File: server/lib/src/database/database_migrator.dart (hunk 2 above)
Comment (klebedev) [inline server/lib/src/database/database_migrator.dart]: Good point. Can't get used to awaits =)

**Pull request merged**